src/cdomains/floatDomain.ml

```diff
@@ -37,18 +37,20 @@ module type FloatDomainBase = sig
 
   val of_const : float -> t
 end
-
 module FloatInterval = struct
   type t = (float * float) option [@@deriving eq, ord, to_yojson]
 
   let norm v = 
-    match v with
-    | Some (low, high) -> 
-      if Float.is_finite low && Float.is_finite high then 
-        if low > high then failwith "invalid Interval"
-        else v
-      else None
-    | _ -> None
+    let normed = match v with
+      | Some (low, high) -> 
+        if Float.is_finite low && Float.is_finite high then 
+          if low > high then failwith "invalid Interval"
+          else v
+        else None
+      | _ -> None
+    in if normed == None then 
+      Messages.warn ~category:Messages.Category.FloatMessage ~tags:[CWE 189; CWE 739] "Could be +/-infinity or Nan";
```

Comment:
i would add "Float" somewhere inside the message e.g. "float value could be ...". If i got this right, this version also warns as soon as the result (or whenever norm is called elsewhere) of the operation is top. Maybe we should incorporate this into the message as well " result of operation could be ...".
Apart from that, keep in mind, that in the future we want to support two options on warning with top inputs and warning on top outputs (i guess that is the current version).